src/Stomp/Transport/Map.php

```diff
@@ -14,25 +14,43 @@
  *
  * @package Stomp
  */
-class Map extends Frame
+class Map extends Message
 {
     public $map;
 
     /**
      * Constructor
      *
-     * @param Frame|string $msg
+     * @param Frame|string $body
      * @param array $headers
+     * @param string $command
      */
-    public function __construct($msg, array $headers = [])
+    public function __construct($body, array $headers = [], $command = 'SEND')
     {
-        if ($msg instanceof Frame) {
-            parent::__construct($msg->command, $msg->headers, $msg->body);
-            $this->map = json_decode($msg->body, true);
+        if ($body instanceof Frame || is_string($body)) {
+            if ($body instanceof Frame) {
+                $headers = $body->headers;
+                $command = $body->command;
+                $body = $body->body;
+            }
+            parent::__construct($body, $headers);
+            $this->command = $command;
+            $this->map = json_decode($body, true);
         } else {
```

Comment:
It's more like a legacy issue here. I'll drop the frame part and break it down to `string` will get parsed by `json_decode` and any `other input` will be wrapped with `json_ecnode`.